server/src/routes/v2/apps.js

```diff
@@ -1,24 +1,76 @@
-const Boom = require('@hapi/boom')
 const Joi = require('@hapi/joi')
+const AppModel = require('../../models/v1/out/App')
+const { AppStatus } = require('../../enums')
+const { getApps } = require('../../data')
+const { convertAppsToApiV1Format } = require('../v1/apps/formatting')
+const { filterAppsBySpecificDhis2Version } = require('../../utils/filters')
+const CustomJoi = require('../../utils/CustomJoi')
 
-const AppModel = require('../../models/v2/out/App')
+const CHANNELS = ['Stable', 'Development', 'Canary']
+const APPTYPES = ['APP', 'DASHBOARD_WIDGET', 'TRACKER_DASHBOARD_WIDGET']
+
+const StringArrayJoi = Joi.extend({
```

Comment:
I'm not sure if we want paging like this and not just "load more"?

Reply:
> I'm not sure if we want paging like this and not just "load more"? I think we can also afford to load more than 12 apps, but I agree 12 makes sense with this approach.

Do you mean something like infinite scroll where apps would be loaded automatically as soon as the user reaches the bottom of the page? My only concern with something like that is that it can 'break' the back button by not restoring the user to their previous scroll position - though tbf I've just realised that since we're not storing pagination details in query params the current state of this PR breaks the back button too. What are your thoughts on using something like [`use-query-params`](https://github.com/pbeshai/use-query-params) to store pagination and filtering state in the URL like how I've done in https://github.com/dhis2/sms-configuration-app/pull/217? 

I picked 12 as [we render 3 app cards per row on smaller screens and 4 otherwise](https://github.com/dhis2/app-hub/blob/b4496df573e8587678e2f7e91f90049bfef5425d/client/src/components/appCards/AppCards.jsx#L131) so the next number we can bump the page size up to is 24. With the UI redesign having a two apps per row layout we will have a wider range of `pageSize` to choose from.

Reply:
Infinite scroll can indeed be annoying when the scroll position is reset, and I guess it would be more complicated to implement - and as we are looking to redesign in in the not too distant future, I think this approach is fine.

Yup I thought about that as well, would be great if we could use the query-params.

Reply:
I've updated the filtering state managent to use query params and I've fixed the subheader back button on the app view page.